### PR TITLE
fixes infinite butter

### LINKED ISSUE #46278: Mount and Blade: Butterlord
## Testmerges:

h

## Reproduction:

You can make infinite butter on a stick, as long as you cradle your butter in-hand while putting rods into it.

1. Pick up your butter.
2. Pick up your rods.
3. Infinitely jam the rods into your in-hand butter because it never disappears. Flood the station with butter. Create the kingdom of Swadia. Feast. Trazelab you bumbling buffoon what have you done

### PR DESCRIPTION
**Pull request by tralezab**
we were not ready
fixes #46278

### CODE REVIEW

**Comment by Bombpants** [comment]
Improve, don’t remove

File: code/modules/food_and_drinks/food/snacks_other.dm [hunk 1]
```diff
@@ -562,6 +562,7 @@
 		R.use(1)
```
Comment (nemvar) [inline code/modules/food_and_drinks/food/snacks_other.dm]: Another thing to consider. R.use() can fail. Borgs can still dispense rods for this even if they are out of metal.

Comment (tralezab) [reply code/modules/food_and_drinks/food/snacks_other.dm]: seems like a borg issue more than a stack issue, but i'll see which side of it is better to fix

Comment (nemvar) [reply code/modules/food_and_drinks/food/snacks_other.dm]: It's kinda fine tbh. use can also be used to subtract larger amounts which can also fail with normal stacks.

File: code/modules/food_and_drinks/food/snacks_other.dm [hunk 2]
```diff
@@ -554,14 +554,18 @@
...
+		if(!R.use(1))//borgs can still fail this if they have no metal
+			to_chat(user, "<span class='warning'>You do not have enough metal to put [src] on a stick!</span>")
+			return ..()
+		to_chat(user, "<span class='notice'>You stick the rod into the stick of butter.</span>")
 		var/obj/item/reagent_containers/food/snacks/butter/on_a_stick/new_item = new(usr.loc)
 		var/replace = (user.get_inactive_held_item() == R)
-		R.use(1)
 		if(!R && replace)
 			user.put_in_hands(new_item)
+		qdel(src)
```
Comment (nemvar) [inline code/modules/food_and_drinks/food/snacks_other.dm]: return TRUE after this so afterattack doesn't get called.

Comment (tralezab) [reply code/modules/food_and_drinks/food/snacks_other.dm]: done

**Comment by JMoldy** [comment]
**Cursed PR**

**Review by Chaznoodles — COMMENTED** [review]
Why does butter on a stick taste like crab

'tastes = list("cream cheese" = 4, "crab" = 3, "crispiness" = 2)'

**Comment by TheVekter** [comment]
> Why does butter on a stick taste like crab
> 
> 'tastes = list("cream cheese" = 4, "crab" = 3, "crispiness" = 2)'

Crab is buttery. I'm more worried about cream cheese being there.

**Pull request merged**